app/code/community/Riskified/Full/Model/Observer.php

```diff
@@ -392,7 +392,7 @@ public function clearRetriesForOrder(Varien_Event_Observer $observer)
         $order = $observer->getOrder();
 
         // Sanity check
-        if (!$order || !$order->getId()) {
+        if (!$order || !$order instanceof Mage_Sales_Model_Order || !$order->getId()) {
```

Comment:
In result that you requested to everything happen in postOrder helper method, I have to pass payload data for events like checkout create and I am using order variable to pass this data. After the request is executed we are dispatching an event and passing order variable to event data.
There is an observer listening on this event and it expects that in order variable there is a order model, in result there was a fatal error -to call a method on a non object. To avoid that I am checking if order is a instance of order model.